### ISSUE TITLE
Add RRule support for amp-date-picker

### ISSUE THREAD

**Issue by caroqliu** [ampproject/amphtml#25244]
Support was removed in #25230 and can be restored by reverting that change and importing RRule.

cc/ @ampproject/wg-ui-and-a11y @nainar

**Comment by rsimha** [comment]
If you run into type check errors while fixing this issue, see [this](https://github.com/ampproject/amphtml/pulls?utf8=%E2%9C%93&q=is%3Apr+sort%3Aupdated-desc+rrule) list of past efforts to correctly import `rrulestr` for use in `extensions/amp-date-picker/0.1/dates-list.js`. It could be that the latest version of `rrule` doesn't have the same problems. Or maybe you'll find examples of how not to do it :)

**Comment by samouri** [comment]
Fixed in https://github.com/ampproject/amphtml/pull/28887
